### PR TITLE
[wait for #1328] [inputlayer] Update input layer with LayerV2 design

### PR DESCRIPTION
**Pull request by kparichay**
Update input layer to upgrade to LayerV2 design

Signed-off-by: Parichay Kapoor <pk.kapoor@samsung.com>

### CODE REVIEW

File: nntrainer/layers/input_layer.cpp [hunk 1]
```diff
@@ -22,15 +22,40 @@
...
+#include <layer_internal.h>
 #include <nntrainer_error.h>
 #include <nntrainer_log.h>
 #include <parse_util.h>
 
 namespace nntrainer {
 
-void InputLayer::setProperty(const PropertyType type,
+void InputLayer::setProperty(const std::vector<std::string> &values) {
+  /// @todo: deprecate this in favor of loadProperties
```
Comment (zhoonit) [inline nntrainer/layers/input_layer.cpp:33]: Maybe add LayerImpl::setProperty(values) to up or down?

Comment (kparichay) [reply nntrainer/layers/input_layer.cpp:33]: up or down?

Comment (zhoonit) [reply nntrainer/layers/input_layer.cpp:33]: Top of this function or bottom of this function :)

Comment (zhoonit) [reply nntrainer/layers/input_layer.cpp:33]: Resolved offline, that there is no problem :)

**Review by zhoonit — APPROVED** [review]
LGTM

**Review by jijoongmoon — APPROVED** [review]
LGTM

**Pull request merged**